### PR TITLE
Automatically create an endpoint alias when logging in

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/gsctl/issues/129

Depends on https://github.com/giantswarm/gsclientgen/pull/22

This PR introduces the automatic addition of an alias for every new endpoint, named after the installation name which is pulled from the API's info endpoint.

Under the following conditions ...

- user has at least two endpoints
- user has just added an endpoint by logging in

... the output now looks like this:

```
gsctl login email@example.com -e https://api.g8s.somealias.eu-central-1.aws.gigantic.io
Password for email@example.com on https://api.g8s.somealias.eu-central-1.aws.gigantic.io:
Endpoint selected: https://api.g8s.somealias.eu-central-1.aws.gigantic.io (somealias)
You are logged in as email@example.com at https://api.g8s.somealias.eu-central-1.aws.gigantic.io.

To switch back to this endpoint, you can use this command:

    gsctl select endpoint somealias

```

### TODO

- [x] make sure an existing alias is not overwritten
- [x] ensure uniqueness of an alias
- [x] switch back to gsclientgen branch master after https://github.com/giantswarm/gsclientgen/pull/22 is merged

### CODE REVIEW

**Comment by marians** [comment]
For a change, this time I didn't ping the usual suspects here for the review.

**Comment by marians** [comment]
FYI I cancelled the review request for @fgimenez (AFK) and asked @xh3b4sd instead.

File: config/config.go [hunk 1]
```diff
@@ -218,6 +246,32 @@ func (c *configStruct) ChooseToken(endpoint, overridingToken string) string {
...
+func (c *configStruct) HasEndpointAlias(alias string) bool {
+	for key := range c.Endpoints {
+		if c.Endpoints[key].Alias == alias {
+			return true
+		}
+	}
+	return false
+}
+
+// GetByAlias performs a lookup by alias and returns the according endpoint URL
```
Comment (MarcelMue) [inline config/config.go]: Godoc function name is wrong.

Comment (marians) [reply config/config.go]: Fixed

**Comment by marians** [comment]
Thanks for your time!

**Pull request merged**